service/history/workflow_task_handler.go

```diff
@@ -345,6 +349,15 @@ func (handler *workflowTaskHandlerImpl) handleCommand(
 		}
 		return nil, err
 	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if historyEvent != nil && command.UserMetadata != nil {
```

Comment:
As synced offline I think ideally the returned historyEvent should not be modified and instead the underlying mutable state and history build can provide a method for mutating added history events. But we can do that later I guess. 
Please add some TODO here though.

Reply:
Will do.

Reply:
Done.